src/renderer/components/query-result-table-virtualized.jsx

```diff
@@ -0,0 +1,388 @@
+                zIndex={999}>
+
+                <div className="draggable-handle"></div>
+
+              </Draggable>
+
+          }
+        
+        })()}
+
+    </div>;
+
+  }
+
+  renderNoRows() {
+    return <div style={{textAlign:"center",fontSize:"16px"}}>No results found</div>;
+  }
+
+  resize(nextProps) {
+    const props = nextProps || this.props;
+    const widthOffset = props.widthOffset;
+    const heigthOffset = props.heigthOffset;
+
+    this.setState({
+      tableWidth: window.innerWidth - (widthOffset + 40),
+      tableHeight: window.innerHeight - (heigthOffset + 225),
+    });
+  }
+
+  renderHeader() {
+    const { rows, rowCount, onCopyToClipboardClick } = this.props;
+    const { tableWidth, tableHeight } = this.state;
+    const styleCopied = {display: this.state.showCopied ? 'inline-block' : 'none'};
+    const styleButtons = {display: this.state.showCopied ? 'none' : 'inline-block'};
+
+    let copyPanel = null;
+    if (rowCount) {
+      copyPanel = (
+        <div className="ui small label" title="Copy as" style={{"float": "right","margin": "3px"}}>
+          <i className="copy icon"></i>
+          <a className="detail" style={styleCopied}>Copied</a>
+          <a className="detail"
+            style={styleButtons}
+            onClick={() => onCopyToClipboardClick(rows, 'CSV')}>CSV</a>
+          <a className="detail"
+            style={styleButtons}
+            onClick={() => onCopyToClipboardClick(rows, 'JSON')}>JSON</a>
+        </div>
+      );
+    }
+
+    return (
+      <div style={{"height":"32px","width":tableWidth}}>
+        <div style={{"marginLeft":"10px","float":"left"}}>Rows: {rowCount}</div>
+        {copyPanel}
+      </div>
+    );
+  }
+
+  getColumnWidth({index}) {
+    var field = this.props.fields[index];
+    if ( field && this.state.columnWidths && this.state.columnWidths[field.name] !== undefined ) {
+      return this.state.columnWidths[field.name];
+    } else if ( this.state.autoColumnWidths && this.state.autoColumnWidths[index] !== undefined ) {
+      return this.state.autoColumnWidths[index];
+    } else {
+      return 50;
+    }
+  }
+
+  render() {
+    const { rowCount, fields, rows } = this.props;
+    const { tableWidth, tableHeight } = this.state;
+    const averageTableCellWidth = tableWidth / fields.length;
+
+    // not completed loaded yet
+    if (!tableWidth) {
+      return null;
+    }
+
+    var j = 0, totalColumnWidths = 0, autoColumnWidths = fields.map(({ name, index }) => this.resolveCellWidth(name, fields, rows, averageTableCellWidth))
+
+    for ( j=0; j < autoColumnWidths.length; j=j+1 ) {
+        totalColumnWidths = totalColumnWidths + autoColumnWidths[j];
+        if ( (j+1) === autoColumnWidths.length && totalColumnWidths < tableWidth ) {
+            totalColumnWidths = totalColumnWidths - autoColumnWidths[j];
+            autoColumnWidths[j] = (tableWidth - totalColumnWidths);
+        }
+    }
+
+    this.state.autoColumnWidths = autoColumnWidths;
```

Comment:
Is not recommend setting the state like this. Should use `this.setState({ autoColumnWidths })`. Also would give a better code readability if we could move the loop above to a function.